hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/ComputeHash.java

```diff
@@ -0,0 +1,954 @@
+
+                    return ret;
+                }
+
+                @Override public DataEntity.Blueprint<?> getReturnType(RelativePath resourceType,
+                                                                       OperationType.Blueprint operationType) {
+                    return members.getReturnType(operationType);
+                }
+
+                @Override public DataEntity.Blueprint<?> getParameterTypes(RelativePath resourceType,
+                                                                           OperationType.Blueprint operationType) {
+                    return members.getParameterTypes(operationType);
+                }
+
+                @Override public DataEntity.Blueprint<?> getConfigurationSchema(ResourceType.Blueprint rt) {
+                    return members.getConfigurationSchema(rt);
+                }
+
+                @Override public DataEntity.Blueprint<?> getConnectionConfigurationSchema(ResourceType.Blueprint rt) {
+                    return members.getConnectionConfigurationSchema(rt);
+                }
+            };
+        }
+
+        static HashableView of(InventoryStructure<?> structure) {
+            return new HashableView() {
+                @Override
+                public DataEntity.Blueprint<?> getConfiguration(RelativePath rootPath,
+                                                                Resource.Blueprint parentResource) {
+                    RelativePath resourcePath = rootPath.modified().extend(SegmentType.r, parentResource.getId())
+                            .get().slide(1, 0);
+
+                    try (Stream<DataEntity.Blueprint<?>> s = structure.getChildren(resourcePath, DataEntity.class)) {
+                        return s.filter(d -> configuration.equals(d.getRole()))
+                                .findFirst().orElse(dummyDataBlueprint(configuration));
+                    }
+                }
+
+                @Override public DataEntity.Blueprint<?> getConfigurationSchema(ResourceType.Blueprint rt) {
+                    RelativePath p = rt.equals(structure.getRoot()) ? RelativePath.empty().get()
+                            : RelativePath.to().resourceType(rt.getId()).get();
+
+                    try (Stream<DataEntity.Blueprint<?>> s = structure.getChildren(p, DataEntity.class)
+                            .filter(d -> configurationSchema.equals(d.getRole()))) {
+                        return s.findFirst().orElse(dummyDataBlueprint(configurationSchema));
+                    }
+                }
+
+                @Override
+                public DataEntity.Blueprint<?> getConnectionConfiguration(RelativePath root,
+                                                                          Resource.Blueprint parentResource) {
+                    RelativePath resourcePath = root.modified().extend(SegmentType.r, parentResource.getId())
+                            .get().slide(1, 0);
```

Comment:
these `.slide(1, 0)` cut off the the very first chunk of the canonical path, right?